components/public-api-server/pkg/apiv1/oidc.go

```diff
@@ -350,3 +354,39 @@ func assertIssuerIsReachable(host string) error {
 	}
 	return nil
 }
+
+func assertIssuerProvidesDiscovery(issuer string) error {
```

Comment:
Should accept context, and pass it into API calls to ensure context concellation